server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -0,0 +1,108 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.Base64
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.boolean.Or
+import eu.timepit.refined.collection.NonEmpty
+import eu.timepit.refined.refineV
+import eu.timepit.refined.string.MatchesRegex
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.jmap.http.UserCredential._
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.apache.james.user.api.UsersRepository
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+import reactor.core.scala.publisher.SFlux
+import reactor.netty.http.server.HttpServerRequest
+
+import scala.compat.java8.StreamConverters._
+
+
+object UserCredential {
+  type BasicAuthenticationHeaderValue = String Refined MatchesRegex["Basic [\\d\\w=]++"]
+  type CredentialsAsString = String Refined MatchesRegex[".*:.*"]
+
+
+  private val logger = LoggerFactory.getLogger(classOf[UserCredential])
+  private val BASIC_AUTHENTICATION_PREFIX: String = "Basic "
+
+  def parseUserCredentials(token: String): Option[UserCredential] = {
+    val refinedValue: Either[String, BasicAuthenticationHeaderValue] = refineV(token)
+
+    refinedValue match {
+      // Ignore Authentication headers not being Basic Auth
+      case Left(_) => None
+      case Right(value) => extractUserCredentialsAsString(value)
+    }
+  }
+
+  private def extractUserCredentialsAsString(token: BasicAuthenticationHeaderValue): Option[UserCredential] = {
+    val encodedCredentials = token.replace(BASIC_AUTHENTICATION_PREFIX, "")
+    val decodedCredentialsString = new String(Base64.getDecoder.decode(encodedCredentials))
+    val refinedValue: Either[String, CredentialsAsString] = refineV(decodedCredentialsString)
+
+    refinedValue match {
+      case Left(errorMessage: String) =>
+        logger.info(s"Supplied basic authentication credentials do not match expected format. $errorMessage")
+        None
+      case Right(value) => toCredential(value)
+    }
+  }
+
+  private def toCredential(token: CredentialsAsString): Option[UserCredential] = {
+    val partSeparatorIndex: Int = token.indexOf(':')
+    val usernameString: String = token.substring(0, partSeparatorIndex)
+    val passwordString: String = token.substring(partSeparatorIndex + 1)
+
+    try {
+      Some(UserCredential(Username.of(usernameString), passwordString))
+    } catch {
+      case throwable: IllegalArgumentException => {
+        logger.info(throwable.getMessage)
+        None
+      }
+      case _ => None
```

Comment:
Here we loose track of an unexpected exception. It's a serious error harming future debugging experience. I would suggest an error log.